### PR TITLE
Add support for using Remote Desktop Gateway servers. 

### LINKED ISSUE #16: Additional field for RDP Gateway server
I connect to many of my RDP hosts via an RDP gateway. You cannot specify the gateway address directly through the command line, but it is possible to build a temporary .RDP connection file every connection. This would allow connection directly to any RDP server. Here is a thread from experts-exchange:

https://www.experts-exchange.com/questions/28342241/Remote-Desktop-Client-CLI-RDP-Gateway.html

Thanks!

### PR DESCRIPTION
**Pull request by mterwoord**
Add support for using Remote Desktop Gateway servers. Works using Rdp+ (See https://www.donkz.nl/)
Add settings for the gateway in same style as putty settings. ("key: value;key2: value2").

Sample:
- rdgateway: my-gateway-hostname-or-ip; useSameCredentials: yes
useSameCredentials is optional.

Specifying the path to Rdp+ should be done in the main settings form.

This PR fixes #16

### CODE REVIEW

**Comment by cristianst85** [comment]
Hi @mterwoord,

Thank you for this PR, but I'll need to make some changes before merging it:

- [ ] Add new dialog that will be used for configuring app paths: _QuickConnect Options_ > _Settings tab_ > _Configure Paths..._ button). This will restore the height of _QuickConnect Options_ dialog to initial size (in a near future I'll be removing _PsPasswd Path_ from _Password Changer_ tab and thus options dialog will shrink even further).
- [ ] Add more test cases for RemoteDesktopArgumentsFormatter to cover the new behaviour;
- [ ] Add test cases for RdpPlusOptionsParser;
- [ ] Code styling.

Maybe:
- [ ] Rename _useSameCredentials_ key to PromptCredentialOnce (case insensitive) and use true/false  values (language independent) instead of yes/no.

What do you think?

P.S. I'll also implement a _native mode_ (using .rdp file) to connect using Remote Desktop Gateway (in a separate PR).

**Comment by mterwoord** [comment]
You mean you are going to make those changes, or you want me to?

regarding the points you mentioned:
- New dialog: Sure, I just thought i'd extend the existing one.
- Tests is always good
- Code styling: I did my best to adhere to your style.

The name of the setting: All for it, just thought, as the user has to enter it manually, we'd better use a simple name.

**Comment by cristianst85** [comment]
If you want you can make those changes, but first you should check the fixes I’ve made to your PR. Didn’t had the time to test them.

Thanks.

Sent from my Windows 10 phone

File: QuickConnectPlugin/ArgumentsFormatters/RemoteDesktopArgumentsFormatter.cs [hunk 1]
```diff
@@ -31,6 +41,23 @@ public class RemoteDesktopArgumentsFormatter : IArgumentsFormatter {
...
             }
 
+            sb.AppendFormat(" /u:\"{0}\"", hostPwEntry.GetUsername());
+            sb.AppendFormat(" /p:\"{0}\"", hostPwEntry.GetPassword());
+
+            if (xUseRdpPlus) {
+                RdpPlusOptions options;
+                if (RdpPlusOptionsParser.TryParse(hostPwEntry.AdditionalOptions, out options)) {
+                    if (!string.IsNullOrEmpty(options.RDGateway)) {
+                        var extraOptions = "gatewayhostname:s:" + options.RDGateway + ", gatewayusagemethod:i:2, gatewaycredentialssource:i:4, gatewayprofileusagemethod:i:1, gatewaybrokeringtype:i:0";
```
Comment (cristianst85) [inline QuickConnectPlugin/ArgumentsFormatters/RemoteDesktopArgumentsFormatter.cs:51]: @mterwoord,

I can't seem to find any references that the `gatewaybrokeringtype` parameter is supported by RDP+. 

https://www.donkz.nl/overview-rdp-file-settings/

Comment (mterwoord) [reply QuickConnectPlugin/ArgumentsFormatters/RemoteDesktopArgumentsFormatter.cs:51]: Those are passed to the /o argument of RDP+, which basically means "pass this on to ms rdp".
What I did was save the rdp file from mstsc without gateway and with gateway, and added those options here.

**Pull request closed without merging**